BinaryRepresentation/BinaryRepresentation.c

```diff
@@ -0,0 +1,221 @@
+        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1},
+        {1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0}
+    };
+    int secondNumberBinTwoByte[AMOUNTTEST][16] =
+    {
+        {1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0},
+        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1},
+        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1}
+    };
+    const int resultTwoByte[AMOUNTTEST][16] =
+    {
+        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1},
+        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0},
+        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1}
+    };
+    int firstNumberBinFourByte[AMOUNTTEST][32] =
+    {
+        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1},
+        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1},
+        {1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0}
+    };
+    int secondNumberBinFourByte[AMOUNTTEST][32] =
+    {
+        {1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0},
+        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1},
+        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1}
+    };
+    const int resultFourByte[AMOUNTTEST][32] =
+    {
+        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1},
+        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0},
+        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1}
+    };
+    for (int i = 0; i < AMOUNTTEST; ++i)
+    {
+        int sumNumberBin[32] = {0};
+        if (size == 16)
+        {
+            bitwiseSum(firstNumberBinTwoByte[i], secondNumberBinTwoByte[i], sumNumberBin, size);
+        }
+        else
+        {
+            bitwiseSum(firstNumberBinFourByte[i], secondNumberBinFourByte[i], sumNumberBin, size);
+        }
+        for (int j = 0; j < size; ++j)
+        {
+            if (size == 16 && sumNumberBin[j] != resultTwoByte[i][j] || size == 32 && sumNumberBin[j] != resultFourByte[i][j])
+            {
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+bool tests()
+{
+    return testForBinToDec() && testForDecToBin() && testForBitwiseSum();
+}
+
+int main()
+{
+    if (!tests())
+    {
+        printf("%s", "Tests failed!");
+        return 0;
+    }
+    setlocale(LC_ALL, "Rus");
+    int firstNumber = 0;
+    printf("%s ", "Введите первое число:");
+    scanf("%d", &firstNumber);
+    int seсondNumber = 0;
+    printf("%s ", "Введите второе число:");
+    scanf("%d", &seсondNumber);
+
+    #define SIZE sizeof(int) * 8
+    const int maxValue = (int)pow(2, SIZE - 1) - 1;
+    const int minValue = (int)pow(2, SIZE - 1) * -1;
```

Comment:
```suggestion
    const int minValue = -(int)pow(2, SIZE - 1);
```
Это как-то более естественно